packages/app/src/domain/exchanges/lifi/query.test.ts

```
@@ -0,0 +1,139 @@
+import { mainnet } from 'viem/chains'
+import { afterEach, describe, expect, test, vi } from 'vitest'
+
+import { BaseUnitNumber, Percentage } from '@/domain/types/NumericValues'
+import { testAddresses } from '@/test/integration/constants'
+import { queryClient } from '@/test/integration/query-client'
+
+import { LiFi } from './lifi'
+import { RealLifiQueryMetaEvaluator } from './meta'
+import { fetchLiFiTxData } from './query'
+
+const dai = testAddresses.token
+const sdai = testAddresses.token2
+const usdc = testAddresses.token3
+const userAddress = testAddresses.alice
+const chainId = mainnet.id
+
+let mockFetch = vi.fn()
+
+describe(fetchLiFiTxData.name, () => {
+  const queryMetaEvaluator = new RealLifiQueryMetaEvaluator({ dai, sdai, usdc })
+  const lifiClient = new LiFi({ chainId, userAddress })
+
+  beforeEach(() => {
+    mockFetch = vi.fn()
+    vi.stubGlobal('fetch', mockFetch)
+  })
+  afterEach(() => {
+    vi.unstubAllGlobals()
+  })
+
+  test('waives fee for dai to sdai conversion', async () => {
+    const amount = BaseUnitNumber(1)
+    const maxSlippage = Percentage(0.005)
+
+    await queryClient
```

i would create a helper function and describe why we ignore errors (because we don't care about response, right?)